### PR TITLE
Dockerfile in katana repo

### PR DESCRIPTION
**Pull request by bkolada**
These are corrections which speeds up a bit setting up environment. This change needs improvements in docker-compose file which will be send in other repo. Also this change requires to maintain requirements file but in my opinion it's small amount which gives us a lot of time savings later.

### CODE REVIEW

**Comment by campos-ddc** [comment]
Definitely very useful to have a docker file somewhere, but I don't think this repo is the right place (since you now create references to buildmaster and buildmaster_configuration which are irrelevant to Katana).

Does anyone else have a suggestion of where we should put this?

**Comment by mariangemarcano** [comment]
@campos-ddc I think it is ok to create a test master called `buildmaster` and we could remove the buildmaster_configuration as we are not using it.

File: master/requirements-dev.txt [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+Jinja2 == 2.8
+sqlalchemy >= 0.6, <= 0.7.9
+sqlalchemy-migrate == 0.7.2
+python-dateutil == 1.5
+autobahn == 0.11.0
+python-ldap
+mysql-python == 1.2.5
+pymysql == 0.7.1
+PyJWT == 1.4.0
+psutil == 4.3.0
```
Comment (bkolada) [inline master/requirements-dev.txt:11]: @campos-ddc in https://github.com/Unity-Technologies/katana/pull/296 mentioned:
> Aren't any of these declared in setup.py? IMO requirements-dev should only contain things that are required for development but not every day use (the every day stuff should be in setup.py)

So let's continue this thread here.

Comment (bkolada) [reply master/requirements-dev.txt:11]: Dockerfile will be used in the developement, won’t it be? Indeed, it’s almost the same as in setup.py and in the current approach we are avoiding any package installation.

Comment (campos-ddc) [reply master/requirements-dev.txt:11]: Not sure if I got what you mean, this is what I would expect:

* If you want to develop new code for Katana, you have to run `setup.py` and `pip install` requirements-dev
* If you just want to start a server, running `setup.py` should be enough.

(doesn't matter if you are using a Dockerfile or not ; `setup.py` will usually be ran when doing a `pip install -e katana/`)

**Comment by bkolada** [comment]
Closed without merge. Here I achieved speeding up `docker-compose up katana` by ±30% of time from command to start parsing buildmasters config. Which sounds good but parsing projects takes much more time than that and it should be solved first. There is one important downside of such solution maintaining requirements.txt will take an effort. It's closed because we decided to create minimal docker container without large buildmaster.

**Pull request closed without merging**